### PR TITLE
Encrypt Backups

### PR DESCRIPTION
**Pull request by mmzeeman**
### Description

Add the possibility to encrypt backups. This makes it safer to store privacy sensitive user content on external backup services. This PR also includes backing up configuration files. These are not downloadable via the admin interface for security reasons.

Please describe here what the PR does.

1) Add possibility to encrypt backups with a password.
2) Backup configuration and security files
3) CLI to decrypt backups with a password.

### Checklist

- [x] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: apps/zotonic_mod_backup/src/models/m_backup.erl [hunk 1]
```diff
@@ -30,6 +30,8 @@ m_get([ <<"admin_panel">> | Rest ], _Msg, Context) ->
     {ok, {m_config:get_value(mod_backup, admin_panel, Context), Rest}};
 m_get([ <<"daily_dump">> | Rest ], _Msg, Context) ->
     {ok, {m_config:get_value(mod_backup, daily_dump, Context), Rest}};
+m_get([ <<"encrypt_backups">> | Rest ], _Msg, Context) ->
+    {ok, {m_config:get_value(mod_backup, encrypt_backups, Context), Rest}};
```
Comment (mworrell) [inline apps/zotonic_mod_backup/src/models/m_backup.erl:34]: Isn't this a boolean?  Like the ones above also should be?

File: apps/zotonic_mod_backup/src/support/z_file_crypto.erl [hunk 2]
```diff
@@ -0,0 +1,175 @@
+%% @author Maas-Maarten Zeeman <maas@channel.me>
+%% @copyright 2024 Maas-Maarten Zeeman
+%% @doc Decrypt and encrypt files.
+
```
Comment (mworrell) [inline apps/zotonic_mod_backup/src/support/z_file_crypto.erl:4]: Add an `@end` here, as otherwise the APL text is added to the `@doc`.

Comment (mworrell) [reply apps/zotonic_mod_backup/src/support/z_file_crypto.erl:4]: Maybe also rename the file to `backup_file_crypto.erl`  (or us it something that would be nice for other modules to use? then we could move it to stdlib or zotonic-support)

**Review by mworrell — APPROVED** [review]

File: apps/zotonic_mod_backup/src/models/m_backup.erl (hunk 1 above)
Comment (mmzeeman) [reply apps/zotonic_mod_backup/src/models/m_backup.erl:34]: Yes

File: apps/zotonic_mod_backup/src/support/z_file_crypto.erl (hunk 2 above)
Comment (mmzeeman) [reply apps/zotonic_mod_backup/src/support/z_file_crypto.erl:4]: Going to rename this.

**Pull request merged**